rust-peer/src/main.rs

```diff
@@ -199,6 +224,39 @@ async fn main() -> Result<()> {
                 SwarmEvent::Behaviour(BehaviourEvent::Kademlia(e)) => {
                     debug!("Kademlia event: {:?}", e);
                 }
+                SwarmEvent::Behaviour(BehaviourEvent::RequestResponse(
+                    request_response::Event::Message { message, .. },
+                )) => match message {
+                    request_response::Message::Request { request, .. } => {
+                        todo!("request_response::Message::Request: {:?}", request);
+                    }
+                    request_response::Message::Response { response, .. } => {
+                        let file_body = response.0;
+                        info!(
+                            "request_response::Message::Response: size:{}",
+                            file_body.len()
+                        );
+                        // TODO: keep this file in memory and provider it via Kademlia
+                    }
+                },
+                SwarmEvent::Behaviour(BehaviourEvent::RequestResponse(
+                    request_response::Event::OutboundFailure {
+                        request_id, error, ..
+                    },
+                )) => {
+                    error!(
+                        "request_response::Event::OutboundFailure for request {:?}: {:?}",
+                        request_id, error
+                    );
+                }
+                SwarmEvent::Behaviour(BehaviourEvent::RequestResponse(
+                    request_response::Event::ResponseSent { peer, request_id },
+                )) => {
+                    debug!(
+                        "request_response::Event::ResponseSent for request {:?} to {:?}",
+                        request_id, peer
+                    );
```

Comment:
If you don't allow any incoming requests, this will never happen so you can ignore the event.

Reply:
good point: cb24cc2